trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/webapp/OutputElementStyleSerializer.java

```diff
@@ -38,11 +40,30 @@ public OutputElementStyleSerializer() {
         super(OutputElementStyle.class);
     }
 
+    /**
+     * Serialize {@link OutputElementStyle} according to TSP, see here for details:
+     * {@link org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.model.OutputElementStyle}
+     */
+    @SuppressWarnings({ "nls", "null" })
     @Override
     public void serialize(@NonNull OutputElementStyle value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
         gen.writeStartObject();
-        gen.writeStringField("parentKey", value.getParentKey()); //$NON-NLS-1$
-        gen.writeObjectField("values", value.getStyleValues()); //$NON-NLS-1$
+        gen.writeStringField("parentKey", value.getParentKey());
+
+        // Verify the type of style values. Make sure only supported types are returned.
+        Map<String, Object> newValues = new HashMap<>();
```

Comment:
Maybe could use startObject/writeField/endObject pattern to write directly without needing copied map, and preserve original map order.

Reply:
Ok. This will avoid looping through the map multiple times. The order doesn't really matter, unless the input map is a LinkedHashMap that preserves the insertion order.